Allow spaces in setClassNames(String...)

### Description of the bug / feature
In HTML you add class names with `<div class="foo bar">`.
In Flow, you add class names with `component.addClassNames(String...)`. 

However, if you type `component.addClassNames("foo bar")`, you will get a `IllegalArgumentException: Class name cannot contain spaces`. You can fix it by changing the code to `component.addClassNames("foo", "bar")`. Writing it with spaces is however quite natural, as that is what it will have in the DOM as well.

I propose that the check for spaces would be removed and that multiple classes separated by space would be allowed and handled correctly.

If it is not possible to pass the string as a whole through the pipes to the client-side,  at least the class name API could automatically do a `classString.split(" ");`.

This is a convenience improvement, that makes developing with Vaadin a little bit natural and faster.

<!-- Explain briefly what is broken or what you want to achieve -->
### Minimal reproducible example
```java
component.addClassNames("foo bar")`
```
### Expected behavior
Component gets in the DOM `class="foo bar"`
### Actual behavior
```
Caused by: java.lang.IllegalArgumentException: Class name cannot contain spaces
	at com.vaadin.flow.internal.nodefeature.ElementClassList$ClassListView.validate(ElementClassList.java:47) ~[flow-server-7.0.0.alpha1.jar:7.0.0.alpha1]
	at com.vaadin.flow.internal.nodefeature.ElementClassList$ClassListView.validate(ElementClassList.java:29) ~[flow-server-7.0.0.alpha1.jar:7.0.0.alpha1]
	at com.vaadin.flow.internal.nodefeature.NodeList$SetView.add(NodeList.java:111) ~[flow-server-7.0.0.alpha1.jar:7.0.0.alpha1]
	at com.vaadin.flow.internal.nodefeature.NodeList$SetView.add(NodeList.java:84) ~[flow-server-7.0.0.alpha1.jar:7.0.0.alpha1]
	at java.base/java.util.AbstractCollection.addAll(AbstractCollection.java:352) ~[na:na]
	at com.vaadin.flow.component.HasStyle.addClassNames(HasStyle.java:142) ~[flow-server-7.0.0.alpha1.jar:7.0.0.alpha1]
	at com.jensjansson.ce.views.main.MainView.createSideMenu(MainView.java:61) ~[classes/:na]
	at com.jensjansson.ce.views.main.MainView.<init>(MainView.java:44) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:45) ~[na:na]
	at java.base/java.lang.reflect.Constructor.newInstance(Constructor.java:490) ~[na:na]
	at org.springframework.beans.BeanUtils.instantiateClass(BeanUtils.java:212) ~[spring-beans-5.3.2.jar:5.3.2]
	... 99 common frames omitted
```
### Versions:
    - Vaadin / Flow version: 20.0.0.alpha3 / 7.0.0.alpha1
    - Java version: 11
    - OS version: Mac OS
    - Browser version (if applicable): - 
    - Application Server (if applicable): - 
    - IDE (if applicable): -

--- a/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
+++ b/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
@@ -15,7 +15,6 @@
  */
 package com.vaadin.flow.component;
 
-import java.util.Arrays;
 import java.util.Set;
 
 import com.vaadin.flow.dom.ClassList;

--- a/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
+++ b/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
@@ -133,7 +133,7 @@ public interface HasStyle extends HasElement {
 
     /**
      * Adds one or more CSS class names to this component. Multiple class names can be
-     * specified by using multiple parameters.
+     * specified by using spaces or by giving multiple parameters.
      *
      * @param classNames the CSS class name or class names to be added to the component
      */
@@ -148,7 +148,7 @@ public interface HasStyle extends HasElement {
 
     /**
      * Removes one or more CSS class names from component. Multiple class names can be
-     * specified by using multiple parameters.
+     * specified by using spaces or by giving multiple parameters.
      *
      * @param classNames the CSS class name or class names to be removed from the component
      */

--- a/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
+++ b/flow-server/src/main/java/com/vaadin/flow/component/HasStyle.java
@@ -139,7 +139,12 @@ public interface HasStyle extends HasElement {
      * @param classNames the CSS class name or class names to be added to the component
      */
     default void addClassNames(String... classNames) {
-        getClassNames().addAll(Arrays.asList(classNames));
+        for (String rawClassName : classNames) {
+            String[] parts = rawClassName.split(" ");
+            for (String part : parts) {
+                getClassNames().add(part);
+            }
+        }
     }
 
     /**
@@ -149,6 +154,11 @@ public interface HasStyle extends HasElement {
      * @param classNames the CSS class name or class names to be removed from the component
      */
     default void removeClassNames(String... classNames) {
-        getClassNames().removeAll(Arrays.asList(classNames));
+        for (String rawClassName : classNames) {
+            String[] parts = rawClassName.split(" ");
+            for (String part : parts) {
+                getClassNames().remove(part);
+            }
+        }
     }
 }

--- a/flow-server/src/test/java/com/vaadin/flow/component/HasStyleTest.java
+++ b/flow-server/src/test/java/com/vaadin/flow/component/HasStyleTest.java
@@ -41,7 +41,7 @@ public class HasStyleTest {
         assertClasses(component, "foo");
         component.addClassName("bar");
         assertClasses(component, "foo", "bar");
-
+        
         // use ClassList
 
         component.getClassNames().add("baz");
@@ -152,6 +152,14 @@ public class HasStyleTest {
         assertClasses(component);
         component.addClassNames("foo", "bar");
         assertClasses(component, "foo", "bar");
+
+        component.removeClassNames("foo bar");
+        assertClasses(component);
+        
+        component.addClassNames("foo bar");
+        assertClasses(component, "foo", "bar");
+       
+
         component.addClassNames("baz1", "baz2");
         assertClasses(component, "foo", "bar", "baz1", "baz2");
     }

